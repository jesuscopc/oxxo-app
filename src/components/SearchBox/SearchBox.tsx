import React from 'react';

const SearchBox = () => {
  return (
    <form className="flex flex-col pt-8 max-w-sm mx-auto">
      <input placeholder="Buscar Productos" className="appearance-none p-2 mb-3 shadow-md" />
      <button
        type="button"
        className="p-2 bg-green-600 font-bold text-white rounded-md ring-green-400"
      >
        Buscar
      </button>
    </form>
  );
};

export default SearchBox;
