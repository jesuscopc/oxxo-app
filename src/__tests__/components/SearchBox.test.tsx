import * as React from 'react';
import { shallow } from 'enzyme';
import SearchBox from 'components/SearchBox/SearchBox';

describe('SUit test SearchBox', () => {
  test('should be render correctly', () => {
    const wrapper = shallow(<SearchBox handleSubmit={() => {}} loading={false} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('button').text()).toBe('Buscar');
  });

  test('should be show loading button', () => {
    const wrapper = shallow(<SearchBox handleSubmit={() => {}} loading={true} />);
    expect(wrapper.find('button').text()).toBe('Loading...');
  });

  test('should be submit form and bubble keyword', () => {
    const wrapper = shallow(<SearchBox handleSubmit={() => {}} loading={true} />);
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });
    expect(wrapper.find('button')).toEqual({});
  });
});
