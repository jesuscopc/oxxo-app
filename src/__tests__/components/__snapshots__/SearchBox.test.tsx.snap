// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SUit test SearchBox should be render correctly 1`] = `
<form
  className="flex flex-col pt-8 max-w-sm mx-auto"
  onSubmit={[Function]}
>
  <input
    className="appearance-none p-2 mb-3 shadow-md"
    placeholder="Buscar Productos"
  />
  <button
    className="p-2 bg-green-600 font-bold text-white rounded-md ring-green-400
          false
          "
    disabled={false}
    type="submit"
  >
    Buscar
  </button>
</form>
`;
